jdbc/src/test/java/com/interface21/jdbc/core/JdbcTemplateTest.java

```diff
@@ -1,5 +1,93 @@
 package com.interface21.jdbc.core;
 
+import static com.interface21.jdbc.core.fixture.UserFixture.DORA;
+import static com.interface21.jdbc.core.fixture.UserFixture.GUGU;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.interface21.jdbc.core.fixture.DataSourceConfig;
+import com.interface21.jdbc.core.fixture.DatabasePopulatorUtils;
+import com.interface21.jdbc.core.fixture.User;
+import com.interface21.jdbc.core.fixture.UserRowMapper;
+import java.util.List;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
 class JdbcTemplateTest {
 
+    JdbcTemplate jdbcTemplate;
+    UserRowMapper userRowMapper;
+
+    @BeforeEach
+    void setup() {
+        jdbcTemplate = new JdbcTemplate(DataSourceConfig.getInstance());
+        userRowMapper = new UserRowMapper();
+        DatabasePopulatorUtils.execute(DataSourceConfig.getInstance());
+    }
+
+    @Nested
+    class Query {
+        @Test
+        void 데이터_여러개를_조회할_수_있다() {
+            // given
+            createUser(GUGU.user());
+            createUser(DORA.user());
+
+            // when
+            List<User> results = jdbcTemplate.query("select * from users", userRowMapper);
+
+            // then
+            assertThat(results).hasSize(2);
+        }
+
+        @Test
+        void 데이터가_없을_땐_빈_리스트를_반환한다() {
+            // when
+            List<User> results = jdbcTemplate.query("select * from users", userRowMapper);
+
+            // then
+            assertThat(results).hasSize(0);
+        }
+    }
+
+    @Nested
+    class Update {
+        @Test
+        void 데이터를_추가할_수_있다() {
+            // given
+            User user = GUGU.user();
+
+            // when
+            int rowsAffected = jdbcTemplate.update("insert into users (account, password, email) values (?, ?, ?)",
+                    user.getAccount(), user.getPassword(), user.getEmail());
+
+            // then
+            assertThat(rowsAffected).isEqualTo(1);
+            assertThat(findAllUser()).hasSize(1);
+            assertThat(findAllUser().get(0).getAccount()).isEqualTo(user.getAccount());
```

Comment:
자바 21에서는 `SequencedCollection`라는 컬렉션 인터페이스이 생기고 추가적인 메서드들이 생겼더라구요
21버전의 List는`SequencedCollection`를 구현하고 있어 
getFirst(), getLast() 같은 새로운 메서드를 활용해 볼 수도 있을 것 같아요 😄

```suggestion
            assertThat(findAllUser().getFirst().getAccount()).isEqualTo(user.getAccount());
```